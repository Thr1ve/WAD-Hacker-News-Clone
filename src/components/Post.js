import React, { PropTypes } from 'react';
import { Link } from 'react-router';

import HTMLContent from './HTMLContent';

// We could easily make these two functions separate components if needed, but for now that's overkill
const getTitle = post => !!post.url ? <a href={post.url} target="_blank">{post.title}</a> : post.title;
const getSubtext = post => !!post.url ? post.url : <HTMLContent html={post.text}/>;

const Post = ({ post }) =>
  <div className="media">
    <div className="media-content">
      <p>
        <strong>{getTitle(post)}</strong>
      </p>
      <small>{getSubtext(post)}</small>
    </div>
    <Link to={`comments/${post.id}`} className="media-right">
      Comments
    </Link>
  </div>;

Post.propTypes = {
  post: PropTypes.object
};

export default Post;
