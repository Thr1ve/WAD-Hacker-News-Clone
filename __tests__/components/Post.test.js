jest.unmock('../../src/components/Post');

import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-addons-test-utils';
import Post from '../../src/components/Post';

describe('Post', () => {
  it('is contained within a div', () => {
    let renderer = TestUtils.createRenderer();
    let post;

    renderer.render(
      <Post post={{ url: "www.google.com", title: "google" }}/>
    );

    post = renderer.getRenderOutput();

    expect(post.type).toEqual('div');
  });
});
