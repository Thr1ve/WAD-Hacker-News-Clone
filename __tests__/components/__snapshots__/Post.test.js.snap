exports[`Post:snapshot should render an object with a url and title property correctly 1`] = `
<div
  className="media">
  <div
    className="media-content">
    <p>
      <strong>
        <a
          href="www.google.com"
          target="_blank">
          google
        </a>
      </strong>
    </p>
    <small>
      www.google.com
    </small>
  </div>
  <a
    className="media-right"
    onClick={[Function bound handleClick]}
    style={Object {}}>
    Comments
  </a>
</div>
`;

exports[`Post:snapshot should render an object without a url property correctly 1`] = `
<div
  className="media">
  <div
    className="media-content">
    <p>
      <strong>
        google
      </strong>
    </p>
    <small>
      <div
        className="content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "a popular search engine"
          }
        } />
    </small>
  </div>
  <a
    className="media-right"
    onClick={[Function bound handleClick]}
    style={Object {}}>
    Comments
  </a>
</div>
`;
